src/main/java/net/neoforged/neoforge/transfer/handlers/templates/InfiniteResourceHandler.java

```
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.templates;
+
+import java.util.Objects;
+import net.neoforged.neoforge.transfer.ResourceHandlerUtil;
+import net.neoforged.neoforge.transfer.handlers.resources.ISingleResourceHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+
+/**
+ * An {@link ISingleResourceHandler} that allows extraction of an unlimited amount of a specified resource.
+ *
+ * @param <T> The type of resource that this storage can accept.
+ */
+public class InfiniteResourceHandler<T extends IResource> implements ISingleResourceHandler<T> {
+    /**
+     * Resource that should be provided infinitely. Mustn't be {@code null}.
+     */
+    public T infinite;
+
+    public InfiniteResourceHandler(T resource) {
+        this.infinite = resource;
```

We should throw an exception if `resource#isEmpty`, as making an infinite resource handler for the empty resource makes no sense. And we should javadoc this constructor and mention that it doesn't support the empty resource.

---

That would be true, but the infinite is mutable. Meaning someone could have an implementation (I do currently) that can set the infinite resource post construction. So cases of an infinite empty, while odd, is valid since there are scenarios of an infinite source was not yet set by the player.

---

Personally I feel like it shouldn't be mutable either... And when a block would want to change it, they can just invalidate their caps, and then return a new instance. It isn't like this should be changing super frequently and be a performance concern...